File: zig/src/agent.c
// Minimal agent runtime implementation for verification
#include <stdint.h>

__attribute__((visibility("default")))
uint32_t execute_agent_default(uint32_t precision, void* entry) {
    (void)precision;
    (void)entry;
    return 0;
}

__attribute__((visibility("default")))
uint32_t execute_agent_explicit(uint32_t precision, void* entry, void* temp_store) {
    (void)precision;
    (void)entry;
    (void)temp_store;
    return 0;
}

File: zig/src/agent.zig
const std = @import("std");

export fn execute_agent_default(precision: u32, entry: *anyopaque) u32 {
    _ = precision;
    _ = entry;
    return 0;
}

export fn execute_agent_explicit(precision: u32, entry: *anyopaque, temp_store: *anyopaque) u32 {
    _ = precision;
    _ = entry;
    _ = temp_store;
    return 0;
}
